### PR TITLE
Update

### PR DESCRIPTION
**Pull request by KwaneleNdenze**

### CODE REVIEW

File: app/helpers/nav_active_link_helper.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module NavActiveLinkHelper
+  def active_link_css(path)
+    current_page?(path) ? 'active' : ''
+  end
+end
```
Comment (thatosmk) [inline app/helpers/nav_active_link_helper.rb:5]: did you change this file?

File: spec/requests/sessions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    get root_path
+    expect(response).to render_template(:index) 
+    
+    sign_out user
+    get root_path
+    expect(response).not_to render_template(:about)
+    expect(response).not_to render_template(:blog)
+    expect(response).not_to render_template(:contact) 
+  end
+end
```
Comment (thatosmk) [inline spec/requests/sessions_spec.rb:17]: use a controller spec and do something like
```ruby
describe 'user_sign_up' do
  subject { post user_path, params: params }
 
  context 'with valid params' do
    let(:params) { attributes_for(:user) }
    it do
      expect{
        subject
      }.to change(User, :count).by(1)
    end
  end
  
  context 'with invalid params' do
  ... # make user params invalid
  end
end
```

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -1,5 +1,22 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+
+  context "valid Factory" do
```
Comment (thatosmk) [inline spec/models/user_spec.rb:5]: say valid user instead

File: app/controllers/users/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  # GET /resource/cancel
+  # Forces the session data which is usually expired after sign
+  # in to be expired now. This is useful if the user wants to
+  # cancel oauth signing in/up in the middle of the process,
+  # removing all OAuth session data.
+  # def cancel
+  #   super
+  # end
+
+  # protected
```
Comment (thatosmk) [inline app/controllers/users/registrations_controller.rb:42]:
```suggestion
```

Comment (thatosmk) [reply app/controllers/users/registrations_controller.rb:42]: remove what you are not using instead of commenting it out

Comment (thatosmk) [reply app/controllers/users/registrations_controller.rb:42]: but only in controllers and models.

File: spec/models/user_spec.rb (hunk 3 above)
Comment (KwaneleNdenze) [reply spec/models/user_spec.rb:5]: Made the change.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+    it "has a valid factory" do
+      expect(build(:user)).to be_valid
+    end
+  end
+
+  context "Validations" do
+    before { create(:user) }
```
[the remark this answers is not in the document]
Comment (KwaneleNdenze) [reply spec/models/user_spec.rb:12]: Made the change.

File: app/controllers/users/registrations_controller.rb (hunk 4 above)
Comment (KwaneleNdenze) [reply app/controllers/users/registrations_controller.rb:42]: Not sure what needs to change here..? Looks pretty much the same to me.

Comment (KwaneleNdenze) [reply app/controllers/users/registrations_controller.rb:42]: Made these changes.

File: app/controllers/pages_controller.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 class PagesController < ApplicationController
+  before_action :authenticate_user!, only: [:dashboard]
```
Comment (thatosmk) [inline app/controllers/pages_controller.rb:2]: I like this

**Pull request merged**